language/move-lang/src/hlir/translate.rs

```diff
@@ -1378,18 +1377,39 @@ fn check_unused_locals(
     locals: &mut UniqueMap<Var, H::SingleType>,
     used: BTreeSet<Var>,
 ) -> BTreeSet<Var> {
+    let signature = context
+        .signature
+        .as_ref()
+        .expect("Signature should always be defined when checking  a function body");
     let mut unused = BTreeSet::new();
-    for (v, _) in locals.iter().filter(|(v, _)| !used.contains(v)) {
+    let mut errors = Vec::new();
+    // report unused locals
+    for (v, _) in locals
+        .iter()
+        .filter(|(v, _)| !used.contains(v) && !v.starts_with_underscore())
+    {
         let vstr = match display_var(v.value()) {
             DisplayVar::Tmp => panic!("ICE unused tmp"),
             DisplayVar::Orig(vstr) => vstr,
         };
-        let msg = format!(
-            "Unused local '{0}'. Consider removing or prefixing with an underscore: '_{0}'",
-            vstr
-        );
-        context.error(vec![(v.loc(), msg)]);
-        unused.insert(v);
+        let loc = v.loc();
+        let msg = if signature.is_parameter(&v) {
```

Comment:
Thoughts on this function? `is_parameter` is `O(n)`

In the past I have convinced myself this is fine since it's user specified parameters (which probably aren't too big?). Just figured I'd bring it up since someone else is now using it :)

Reply:
I agree with your reasoning. This should be as fast as any fancier scheme for realistic #'s of parameters.